src/RequestInformation.cs

```diff
@@ -174,7 +172,7 @@ public void AddQueryParameters<T>(T source)
             }
         }
 
-        private static object ExpandArray(Array collection)
+        private static string[] ExpandArray(Array collection)
         {
             var passedArray = new string[collection.Length];
```

Comment:
I believe the toString at line 182 should be removed and the return type changed to object[]

Reply:
I just tested that change locally and it breaks the `DoesNotExpandSecondLayerArrays` test with the following error:

> Message: 
>   System.ArgumentException : Illegal class passed as substitution, found System.Int32[]
>
> Stack Trace: 
>   UriTemplate.convertNativeTypes(Object value)
>   UriTemplate.AddExpandedValue(String prefix, Object value, StringBuilder result, Int32 maxChar, Boolean replaceReserved)
>   UriTemplate.AddValue(Nullable`1 op, String token, Object value, StringBuilder result, Int32 maxChar)
>   UriTemplate.AddListValue(Nullable`1 op, String token, IList value, StringBuilder result, Int32 maxChar, Boolean composite)
>   UriTemplate.ExpandToken(Nullable`1 op, String token, Boolean composite, Int32 maxChar, Boolean firstToken, IReadOnlyDictionary`2 substitutions, StringBuilder result, Int32 col)
>   UriTemplate.ExpandImpl(String str, IReadOnlyDictionary`2 substitutions)
>   UriTemplate.Expand(String template, IReadOnlyDictionary`2 substitutions)
>   RequestInformation.get_URI() line 103
>   RequestInformationTests.DoesNotExpandSecondLayerArrays() line 659
RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)

Reply:
I think this is a difference in expectation (generating a URL that is not useful vs throwing an error). It may make sense to update the Test to expect the error as second layer arrays are not supported really. Thoughts?

Reply:
yep, I've just updated this to that so we can fast track a release before the generator's release